### ISSUE TITLE
Using albef to do image tagging

### ISSUE THREAD

**Issue by cs20s030** [salesforce/LAVIS#3]
Hi, i want to do image tagging using albef model. I have writen the code for that but the top one tags are way off. Could you help we regarding this
def calculate_pred(img):
    raw_image = Image.open(img).convert("RGB")
    image = vis_processors["eval"](raw_image).unsqueeze(0).to(device)
    text = [[txt_processors["eval"](template+x)] for x in occ_list] #occ_list is list of occupation and template = "photo of a "
    sample = [{"image": image, "text_input": text_input} for text_input in text]
    img_feats = model.extract_features(sample[0], mode="image").image_embeds_proj[:,0,:]
    img_feats = img_feats/img_feats.norm()
    text_feats = torch.cat([model.extract_features(sample[i], mode="text").text_embeds_proj[:,0,:] / model.extract_features(sample[i], mode="text").text_embeds_proj[:,0,:].norm() for i in range(len(sample))])
    _, index = torch.max(img_feats @ text_feats.T, dim=1)
    return(occ_list[index.tolist()[0]])

**Comment by dxli94** [comment]
Hi @cs20s030, 

Thanks a lot for your interest in our work.

We have looked into the issue. It turned out that there was a bug when loading pre-trained weights for ALBEF feature extractor, which leads to text encoder weights not properly loaded. Please see [this commit](9965e3418b45324b0600b452ec6f641a09481421) for details.

Also, text_embeds_proj and image_embeds_proj are normalized features in the low-dimension space. So don't need to normalize, again (although it doesn't make a difference as norm=1).

Can you pull and let us know how it works now?

Thanks.
